### PR TITLE
Support layer norm and batch norm in classification and regression decoders.

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
GraphStorm decoders do not support normalization for their neural network layers, such as batch norm and layer norm. 

*Description of changes:*
This PR changes the APIs of GraphStorm classification and regression decoders to add a new argument ``norm`` in each ``__init__`` function. The argument allows users to define the normalization type of the neural network layers of these decoders. 
Note: only ``MLPEFeatEdgeDecoder`` implements layer/batch norm between its layers.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/config/argument.py [hunk 1]
```diff
@@ -690,6 +690,7 @@ def verify_arguments(self, is_train):
             _ = self.max_grad_norm
             _ = self.grad_norm_type
             _ = self.gnn_norm
+            _ = self.decoder_norm
```
[the remark this answers is not in the document]
Comment (classicsong) [reply python/graphstorm/config/argument.py:693]: I think it is better to set it separately as users can set gnn_norm to 'layer' will keep decoder_norm as none.

**Review by zhjwy9343 — APPROVED** [review]
LGTM
